src/main/kotlin/com/demonwav/mcdev/platform/mixin/config/reference/CompatibilityLevel.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.platform.mixin.config.reference
+
+import com.demonwav.mcdev.platform.mixin.util.MixinConstants.Classes.COMPATIBILITY_LEVEL
+import com.demonwav.mcdev.util.reference.InspectionReference
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.psi.JavaPsiFacade
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiEnumConstant
+import com.intellij.psi.PsiReference
+import com.intellij.psi.PsiReferenceBase
+import com.intellij.psi.PsiReferenceProvider
+import com.intellij.util.ArrayUtil
+import com.intellij.util.ProcessingContext
+
+object CompatibilityLevel : PsiReferenceProvider() {
+
+    override fun getReferencesByElement(element: PsiElement, context: ProcessingContext): Array<PsiReference> = arrayOf(Reference(element))
+
+    private class Reference(element: PsiElement) : PsiReferenceBase<PsiElement>(element), InspectionReference {
+
+        override val description
+            get() = "compatibility level '%s'"
+
+        override val unresolved
+            get() = resolve() == null
+
+        override fun resolve(): PsiElement? {
+            val compatibilityLevel = findCompatibilityLevel(element)
+            return compatibilityLevel?.findFieldByName(value, false)?.takeIf { it is PsiEnumConstant }
```

Comment:
It could change when updating Mixin. I don't think it's very expensive so I'm not sure if caching it is worth it (also need to make sure we don't cause memory leaks by keeping references to invalidated psi elements)